urbanairship-chat-react-native/ios/AirshipChatModule.m

```diff
@@ -0,0 +1,50 @@
+#import "AirshipChatModule.h"
+#import "ConversationUpdatedDelegate.h"
+@import Airship;
+@implementation AirshipChatModule
+
+RCT_EXPORT_MODULE()
+
+RCT_EXPORT_METHOD(openChat) {
+    [[NSOperationQueue mainQueue] addOperationWithBlock:^ {
+        [[UAirshipChat shared] openChat];
+    }];
+}
+
+RCT_EXPORT_METHOD(sendMessage:(NSString *)message) {
+    [[UAirshipChat shared].conversation sendMessage:message];
+}
+
+RCT_EXPORT_METHOD(sendMessageWithAttachment:(NSString *)message
+                  :(NSURL *)attachmentUrl) {
+    [[UAirshipChat shared].conversation sendMessage:message attachment:attachmentUrl];
+}
+
+RCT_REMAP_METHOD(getMessages,
+                 getMessages_resolver:(RCTPromiseResolveBlock)resolve
+                 rejecter:(RCTPromiseRejectBlock)reject) {
+
+    [[UAirshipChat shared].conversation fetchMessagesWithCompletionHandler:^(NSArray<UAChatMessage *> *messages) {
+        NSMutableArray *mutableMessages = [messages mutableCopy];
+        for (UAChatMessage *message in messages) {
+            NSMutableDictionary *messageInfo = [NSMutableDictionary dictionary];
+            [messageInfo setValue:message.messageID forKey:@"messageId"];
+            [messageInfo setValue:message.text forKey:@"text"];
+            [messageInfo setValue:message.timestamp forKey:@"createdOn"];
```

Comment:
message.timestamp on iOS is a Date, not a number. you may need to convert to milliseconds:

`message.timestamp.timeIntervalSince1970 * 1000.0`